src/plugins/memoryvalue/memoryvalue.c

```diff
@@ -12,63 +12,63 @@
 #include <kdberrors.h>
 #include <kdbhelper.h>
 #include <kdbtypes.h>
-#include <regex.h>
 #include <stdio.h>
 #include <stdlib.h>
 
-
-static kdb_unsigned_long_long_t is_valid_key (Key * key)
+static char * deblank (char * input)
 {
-
-	const Key * meta = keyGetMeta (key, "check/memoryvalue");
-	const char * pattern = "^[0-9]+[ ]*(KB|MB|GB|TB|PB|B)$";
-	const char * value = keyString (key);
-	regmatch_t offsets;
-	regex_t regex;
-	int compile_failure;
-	int match;
-	if (!meta)
+	kdb_unsigned_long_long_t i, j;
+	char * output = input;
+	for (i = 0, j = 0; i < strlen (input); i++, j++)
 	{
-		return 0;
+		if (input[i] != ' ')
+			output[j] = input[i];
+		else
+			j--;
 	}
+	output[j] = 0;
+	return output;
+}
 
-	compile_failure = regcomp (&regex, pattern, REG_NOSUB | REG_EXTENDED | REG_NEWLINE);
+static kdb_unsigned_long_long_t isValidKey (Key * key)
+{
 
-	if (compile_failure)
-	{
-		return 0;
-	}
+	const char * value = keyString (key);
+	// else we manipulate the original
+	char * tempval = elektraStrDup (value);
```

Comment:
You forgot to `elektraFree(tempval)`. If you don't modify the value (i.e. `deblank` just skips all spaces and returns the new position in the string) you won't have to `elektraStrDup` at all.